load.go

```diff
@@ -15,25 +15,37 @@ func Overload() error {
 }
 
 func OptLookupGit() {
+	d.mu.Lock()
 	d.opts.lookupGit = true
+	d.mu.Unlock()
```

Comment:
move to a function called acquire locks, and which returns release function which you can use it along with defer.